src/scripts/danish-legal/create_completeness_detection.py

```diff
@@ -0,0 +1,299 @@
+    dataset_id = "EuroEval/legal-completeness-detection"
+    try:
+        api = HfApi()
+        api.delete_repo(dataset_id, repo_type="dataset")
+    except HTTPError:
+        pass
+
+    # Upload dataset
+    dataset.push_to_hub(dataset_id, private=True)
+
+
+def parse_contract_into_sections(markdown_text: str) -> List[str]:
+    """Parse markdown into sections, excluding subsections with bullet points.
+
+    Args:
+        markdown_text: The markdown text of the contract
+
+    Returns:
+        A list of sections
+    """
+    sections = re.split(
+        r"(?=^## \d+\.\s+(?!.*\n\n-))", markdown_text, flags=re.MULTILINE
+    )
+    return [section.strip() for section in sections if section.strip()]
+
+
+def generate_unique_samples(
+    contract_sections: List[str], not_required_indices: List[int], num_samples: int
+) -> List[Dict]:
+    """Generate unique contract samples."""
+    samples: List[Dict] = []
+    seen_contracts = set()
+    attempts = 0
+    max_attempts = num_samples * 10
+
+    while len(samples) < num_samples and attempts < max_attempts:
+        attempts += 1
+
+        # Generate contract
+        if random.random() < COMPLETE_CONTRACT_PROBABILITY:
+            contract = _create_complete_contract(
+                contract_sections, not_required_indices
+            )
+            target_text = (
+                "Kontrakten er komplet og indeholder alle nødvendige elementer."
+            )
+        else:
+            contract = _create_incomplete_contract(
+                contract_sections=contract_sections,
+                not_required_indices=not_required_indices,
+            )
+            target_text = (
+                "Kontrakten mangler at beskrive følgende emner: "
+                f"{', '.join(contract['missing_categories'])}"
+            )
+
+        # Check if this contract is unique
+        contract_text = contract["contract_text"]
+        if contract_text not in seen_contracts:
+            seen_contracts.add(contract_text)
+
+            sample = {
+                "text": f"{SYSTEM_PROMPT}\n\n{contract_text}",
+                "target_text": target_text,
+                "is_complete": contract["is_complete"],
+                "missing_categories": contract["missing_categories"],
+            }
+
+            samples.append(sample)
+            logger.info(f"Generated unique sample {len(samples)}/{num_samples}")
+        else:
+            logger.debug(f"Duplicate contract detected, skipping (attempt {attempts})")
+
+    logger.info(f"Generated {len(samples)} unique samples out of {attempts} attempts")
+    return samples
+
+
+def _renumber_section_title(section: str, new_number: int) -> str:
+    """Renumber a section's title from ## X. Title to ## new_number. Title."""
+    return re.sub(r"^## \d+\.", f"## {new_number}.", section)
+
+
+def _sample_missing_categories(min_missing: int = 1, max_missing: int = 3) -> List[str]:
+    """Sample 1-3 categories to remove."""
```

Comment:
The default values shouldn't be used in the docstring, as it will inevitably lead to errors.